src/org/mozilla/javascript/NativeArray.java

```diff
@@ -1962,6 +1970,18 @@ private static Object js_copyWithin(
         return thisObj;
     }
 
+    private static Object js_at(Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        Scriptable o = ScriptRuntime.toObject(cx, scope, thisObj);
+        Object targetArg = (args.length >= 1) ? args[0] : Undefined.instance;
```

Comment:
ScriptRuntime has overloaded variants of toInteger that might be a better fit here, like `public static double toNumber(Object[] args, int index)`